### PR TITLE
Hoverboard - Traveling merchant donor item.

### PR DESCRIPTION
**Pull request by warior4356**
permission
This item is for Travelling merchant as his fluff donator item. The hover board is simply a modified wheelchair that can be picked up and stored in a backpack slot. The item has already gone through review and was greenlit by Neca and Free to be made. It travels at walking speed but the downside is that it can not be used like a wheelchair to aid movement if the user is without feet and injured. The item does zero damage in combat.

🆑 warior4356
add: New donor item Hover board.
/🆑

### CODE REVIEW

**Comment by TravellingMerchant** [comment]
Well, here it is at last! Many thanks to warrior for all his time and coding!

**Comment by Necaladun** [comment]
*flip *ollie *rad360grind

**Comment by Anticept** [comment]
Connnnfliiiictttsss

**Comment by Kyep** [comment]
@Anticept Con'fixeeeeeed

**Comment by TravellingMerchant** [comment]
Gnarly. Surf's up duuuuudes!

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -697,3 +697,98 @@
...
+		var/mob_delay = thedriver.movement_delay()
+		if(mob_delay > 0)
+			calculated_move_delay += mob_delay
+
+		if(ishuman(buckled_mob))
+			var/mob/living/carbon/human/driver = user
+			var/obj/item/organ/external/l_foot = driver.get_organ("l_foot")
+			var/obj/item/organ/external/r_foot = driver.get_organ("r_foot")
+			if((!l_foot || (l_foot.status & ORGAN_DESTROYED)) && (!r_foot || (r_foot.status & ORGAN_DESTROYED)))
+				return 0 // No hands to drive your chair? Tough luck!
```
Comment (KasparoVy) [inline code/modules/customitems/item_defines.dm]: These are feet not hands

Comment (warior4356) [inline code/modules/customitems/item_defines.dm]: I stole the code from wheelchairs, its just a comment. Chill. Its a donor item.

Comment (warior4356) [inline code/modules/customitems/item_defines.dm]: If you want me to change it I will. But I dont think its that big a deal.

Comment (SamHPurp) [inline code/modules/customitems/item_defines.dm]: For code immurshuns, change it

Comment (warior4356) [inline code/modules/customitems/item_defines.dm]: There I fixed it.

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -697,3 +697,98 @@
...
+
+/obj/structure/stool/bed/chair/wheelchair/fluff/traveling_merchant/MouseDrop(over_object, src_location, over_location)
+    ..()
+    if(over_object == usr && Adjacent(usr) && (in_range(src, usr) || usr.contents.Find(src)))
+        if(!ishuman(usr))
+            return
+        if(buckled_mob)
+            return 0
+        usr.visible_message("<span class='notice'>[usr] collapses \the [src.name].</span>", "<span class='notice'>You collapse \the [src.name].</span>")
+        new/obj/item/weapon/fluff/traveling_merchant(get_turf(src))
```
Comment (FlattestGuitar) [inline code/modules/customitems/item_defines.dm]: Should transfer fingerprints.

**Comment by TravellingMerchant** [comment]
Ready to merge?

**Comment by Fox-McCloud** [comment]
This is a ton of copy-pasta from the wheelchair; the `relaymove` proc is almost entirely copy-pasted.

Secondly, most of this code uses spaces instead of tabs; a huge no go.

Lastly, you double the speed on this over the wheelchair, making it super super fast. You even removed the detrimental part of slowdowns...so not only do you even have less slowdowns that before, but, again, you can travel super fast with this. Additionally, this makes you immune to slipping and slap lube. This item is a powergamer's dream for having--it provides absolutely no disadvantages at all and just stacks on mechanical benefits....and this is all for a fluff item. Fluff items aren't pay to win and shouldn't impact game balance ***at alll***.

**Comment by TravellingMerchant** [comment]
It shouldn't be faster than normal running speed. That does need to be fixed.

**Comment by taukausanake** [comment]
You know, I'm surprised hoverboards were not made before this fluff PR. Kind of a missed opportunity for something really cool but as Fox pointed out, broken in regards for slipping. It makes sense unless it's like Back to the Future where the hoverboard doesn't work on water. That would be interesting to see the board stop on lube and water spots so when you get off it you slip

**Comment by theColdflame** [comment]
a fully assembled wheelchair in a backpack is an objective advantage, even if the other stuff is brought in-line with it

tbh this would be better as a general feature than a fluff item imo regardless

**Comment by Necaladun** [comment]
Balance wise fox points out a few things there I was unaware of.

I'd say it's prolly better to just have it as a pair of shoes.

**Comment by Necaladun** [comment]
This is turning out pretty complicated. A wheelchair itself has a bunch of issues, and a faster wheelchair type thing that can be assembled out of the backpack is going way beyond the intent for a fluff item, which is meant to be a simple reskin or something like a jacket or a hat. Hats were really the original intent.

I could see this as a resprite for shoes, or perhaps as a resprite of hoverboards if they ever were done, but as things are now this appears too fatally flawed.

**Pull request closed without merging**